packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -442,6 +442,16 @@ export function getEnvForPackageManager({
       newEnv.PATH = `/node16/bin-npm7:${env.PATH}`;
       console.log('Detected `package-lock.json` generated by npm 7...');
     }
+  } else if (cliType === 'pnpm') {
+    if (
+      typeof lockfileVersion === 'number' &&
+      lockfileVersion === 5.4 &&
+      (nodeVersion?.major || 0) > 12
+    ) {
+      // Ensure that pnpm 7 is at the beginning of the `$PATH`
+      newEnv.PATH = `/pnpm7/pnpm:${env.PATH}`;
```

Comment:
```suggestion
      newEnv.PATH = `/pnpm7:${env.PATH}`;
```

Wouldn't it be the directory only?

Might be good to add the e2e test to confirm like you didn't for the other pnpm tests.

Reply:
Oh maybe - the executable itself is at `pnpm7/pnpm` but i guess for path you just need the directory not the executable. I will add an integration test now

Reply:
this doesn't work if its not the full executable path